src/ui/treeview/treeview.ts

```diff
@@ -14,27 +17,60 @@ export default class TmcMenuTree {
     constructor(viewId: string) {
         this.treeDP = new TmcMenuTreeDataProvider();
         vscode.window.registerTreeDataProvider(viewId, this.treeDP);
+
+        this.visibility = new Visibility();
+        this.nextId = 1;
     }
 
     /**
      * Register an action to be shown in the action treeview.
-     * @param label An unique label, displayed in the treeview
+     *
+     * @param label A label, displayed in the treeview
      * @param onClick An action handler
-     * @param visible Determines whether the action should be visible in the treeview
+     * @param groups Determines when the action should be visible in the treeview
      */
-    public registerAction(label: string, id: string, onClick: () => void, visible: boolean) {
+    public registerAction(label: string, groups: string[], onClick: () => void) {
+        const id = (this.nextId++).toString();
+
+        // Use internal classes
+        this.visibility.registerAction(id, groups);
+        this.treeDP.registerAction(label, id, onClick, this.visibility.getVisible(id));
+    }
+
+    /**
+     * Register a visibility group for the action treeview
+     * @param group Name of the group
+     * @param visible Whether the group should start as active or not
+     */
+    public registerVisibilityGroup(group: string, visible?: boolean) {
         // Use internal class
-        this.treeDP.registerAction(label, id, onClick, visible);
+        this.visibility.registerGroup(group, visible ? visible : false);
```

Comment:
Is "visible?" necessary, as tmc.isAuthenticated() should return true or false and never undefined?

Reply:
There will be other groups, and this allows groups to be created invisible by default without passing the visibility parameter at all

Reply:
Sure thing, didn't think of that. We probably want later on visibility groups that are also shown to people not logged in.